commit fb9d564b501d614b5383f4ca631d78cf23ae6d98
Author: bertin <macportsuser@macports.org>
Date:   Fri Oct 10 19:08:57 2025 +0200

    /opt/local/linux-ports/audio/audacity/files/patch-rearrange-control-buttons.diff

diff --git a/src/toolbars/ControlToolBar.cpp b/src/toolbars/ControlToolBar.cpp
index 8fa1193e8..355da6703 100644
--- a/src/toolbars/ControlToolBar.cpp
+++ b/src/toolbars/ControlToolBar.cpp
@@ -327,6 +327,14 @@ void ControlToolBar::UpdatePrefs()
    ToolBar::UpdatePrefs();
 }
 
+#if defined(USE_AQUA_THEME) && !defined(__WXMAC__)
+// add some additional spacing to distribute the control buttons in
+// the toolbar, which is given an oversized minimal width
+#define ADD_SPACING(dx,dy) mSizer->Add((dx),(dy))
+#else
+#define ADD_SPACING(dx,dy) /**/
+#endif
+
 void ControlToolBar::ArrangeButtons()
 {
    int flags = wxALIGN_CENTER | wxRIGHT;
@@ -342,6 +350,7 @@ void ControlToolBar::ArrangeButtons()
 
    // Start with a little extra space
    mSizer->Add( 5, 55 );
+   ADD_SPACING( 17, 0 );
 
    // Establish correct tab key sequence with mLoop last
    mPause->MoveBeforeInTabOrder( mLoop );
@@ -352,11 +361,17 @@ void ControlToolBar::ArrangeButtons()
    mRecord->MoveBeforeInTabOrder( mLoop );
 
    mSizer->Add( mPause,  0, flags, 2 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mPlay,   0, flags, 2 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mStop,   0, flags, 2 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mRewind, 0, flags, 2 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mFF,     0, flags, 10 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mRecord, 0, flags, 10 );
+   ADD_SPACING( 7, 0 );
    mSizer->Add( mLoop, 0, flags, 5 );
 
    // Layout the sizer
